rust/lance-core/src/error.rs

```diff
@@ -111,6 +111,11 @@ pub enum Error {
         minor_version: u16,
         location: Location,
     },
+    #[snafu(display("DataFusion inner error: {source}, {location}"))]
+    DataFusionInnerError {
```

Comment:
currently I think all datafusion errors are converted to execution error already?

```rust
#[cfg(feature = "datafusion")]
impl From<Error> for datafusion_common::DataFusionError {
    #[track_caller]
    fn from(e: Error) -> Self {
        Self::Execution(e.to_string())
    }
}
```